python/ray/experimental/dag/py_obj_scanner.py

```diff
@@ -49,6 +49,13 @@ def __init__(self):
         from ray.serve.pipeline.deployment_node import DeploymentNode
         from ray.serve.pipeline.deployment_method_node import DeploymentMethodNode
         from ray.serve.pipeline.deployment_function_node import DeploymentFunctionNode
+        from ray.serve.deployment_executor_node import DeploymentExecutorNode
+        from ray.serve.deployment_method_executor_node import (
+            DeploymentMethodExecutorNode,
+        )
+        from ray.serve.deployment_function_executor_node import (
+            DeploymentFunctionExecutorNode,
+        )
```

Comment:
seems bad to me that we need to register these in the core DAG code... if external parties wanted to add a DAG implementation, would they also need to do this? Is there a way we can define an interface to register these instead of hard-coding them? cc @ericl if you have thoughts